src/lua/tests.rs

```diff
@@ -0,0 +1,31 @@
+//! Lua thread test.
+//! Spawns a lua thread!
+
+use std::time::Duration;
+use std::thread;
+
+use super::*;
+
+#[test]
+fn big_lua_test() {
+    println!("Testing lua thread...");
+    super::init();
+    thread::sleep(Duration::from_secs(5));
```

Comment:
Okay, upgrading the Lua API to smooth out restarts would include some work from `async-lua`.